server/registry/internal/storage/save.go

```diff
@@ -94,14 +91,3 @@ func (c *Client) save(ctx context.Context, v interface{}) error {
 	}).Create(v).Error
 	return grpcErrorForDBError(ctx, errors.Wrapf(err, "save %#v", v))
 }
-
-func (c *Client) saveWithMask(ctx context.Context, v interface{}, fieldMask *fieldmaskpb.FieldMask) error {
-	op := c.db.WithContext(ctx).
-		Select(fieldMask.GetPaths()).
-		Clauses(clause.Returning{})
-	err := op.Save(v).Error
-	if err == nil && op.RowsAffected == 0 {
-		err = status.Errorf(codes.NotFound, "%s not found in database", v)
-	}
```

Comment:
This had a surprising side-effect: if two identical update calls to the same resource were made sequentially, the second call would affect no rows and falsely report that the resource was not found.